### PR TITLE
Add various exports.

### PR DESCRIPTION
**Pull request by TimonPost**
Adds various missing exports.

### CODE REVIEW

**Comment by tobz** [comment]
Sorry, I realized there was a commit that I had meant to push this morning, and it did some of the stuff you have in your PR, but also moved some exports around.  Just pushed it now so that's the cause of the conflict here.

**Comment by TimonPost** [comment]
Cool then I close this one, wasn't to much work. Have you also exposed the: distribution, distribution builder and the sanitize functions? I have not seen those yet. Would like to be able to use those in my project. An other question, when are you planning to release those changes?

**Comment by tobz** [comment]
It doesn't expose anything related to the Prometheus exporter, so feel free to reopen this: I have a few thoughts on how I'd want the exports structured.

I'm happy to cut a new release of `metrics-util` whenever but if you also want some new exports in `metrics-exporter-prometheus`, then I'd rather get this/whatever PR merged in first so I can do the releases all at once.

**Comment by TimonPost** [comment]
Just updated the PR.

Would be really nice if there is a new release within this week. As im working on some project and hope to open a PR for it this week. Otherwise I will have to copy logic which feels a bit bad.

File: metrics-exporter-prometheus/src/lib.rs
Comment (tobz) [inline metrics-exporter-prometheus/src/lib.rs:104]: I think what I'd want to do here is move all of these `sanitize_*` free functions into a new module called `rendering` at the top level, and then expose the module publicly.

That gives us a place to expose more of the rendering code in the future without cluttering the top level of the crate, in terms of what the docs end up looking like.

**Review by tobz — CHANGES_REQUESTED** [review]

File: metrics-exporter-prometheus/src/lib.rs
Comment (TimonPost) [reply metrics-exporter-prometheus/src/lib.rs:104]: Good idea. I can do that.

Comment (TimonPost) [reply metrics-exporter-prometheus/src/lib.rs:104]: Maybe we can call it formatting @tobz ?

Comment (tobz) [reply metrics-exporter-prometheus/src/lib.rs:104]: Yeah, I'm fine with `formatting`.

Comment (TimonPost) [reply metrics-exporter-prometheus/src/lib.rs:104]: I also moved the write_line functions to this module. They all have the same thing in common, formatting strings and sanitizing them for Prometheus.

**Comment by tobz** [comment]
Looks like this needs `cargo fmt` run on it, and the test failures need to be addressed.

**Comment by TimonPost** [comment]
Pushed to fix some of the compile errors. Locally it works now, CI is waiting for your approval.

**Comment by TimonPost** [comment]
For now, I did not expose:

```
invalid_label_key_character,
invalid_label_key_start_character,
invalid_metric_name_character, 
invalid_metric_name_start_character, 
sanitize_label_value_or_descpiption
```

Since they did not seem very relevant and were mostly used by the relevant functions:

```
sanitize_description
sanitize_label_value
sanitize_label_key
sanitize_metric_name
write_metric_line
write_type_line
write_help_line
key_to_parts
```

**Comment by tobz** [comment]
This is released in `metrics-exporter-prometheus@0v0.9.0`.

Thanks again for your contribution!

**Comment by TimonPost** [comment]
Awesome @tobz !

**Pull request merged**
